service/src/main/java/bio/terra/axonserver/service/wsm/WorkspaceManagerService.java

```diff
@@ -0,0 +1,39 @@
+package bio.terra.axonserver.service.wsm;
+
+import bio.terra.axonserver.app.configuration.WsmConfiguration;
+import bio.terra.workspace.api.ResourceApi;
+import bio.terra.workspace.client.ApiClient;
+import bio.terra.workspace.client.ApiException;
+import bio.terra.workspace.model.ResourceDescription;
+import java.util.UUID;
+import javax.ws.rs.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class WorkspaceManagerService {
+  private final WsmConfiguration wsmConfig;
+
+  @Autowired
+  public WorkspaceManagerService(WsmConfiguration wsmConfig) {
+    this.wsmConfig = wsmConfig;
+  }
+
+  private ApiClient getApiClient(String accessToken) {
+    ApiClient apiClient = getApiClient();
+    apiClient.setAccessToken(accessToken);
+    return apiClient;
+  }
+
+  private ApiClient getApiClient() {
+    return new ApiClient().setBasePath(wsmConfig.basePath());
+  }
+
+  public ResourceDescription getResource(String accessToken, UUID workspaceId, UUID resourceId) {
```

Comment:
I see that you are handing up the resource and then switching on resource type. Whenever I see a `switch` I ask, should this be done with a class hierarchy or an interface? The answer is not always "yes", but it is worth thinking about.

For example, you might make an interface that has methods like `isAFileResource`, `getFileContents` or `getFileStream` (or some such). Then you have a place to specialize across different clouds and resources, rather than doing a switch dispatch.

Just to be clear, I am not asking you to do this; only suggesting you consider it and keep it in mind in the future.
Not sure it is worth it for what you have now, but something to keep in mind as the

Reply:
I see, keeping these patterns in mind.